src/main/kotlin/utils/StringMatchingUtils.kt

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import com.intellij.notification.NotificationDisplayType
+import com.intellij.notification.NotificationGroup
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.project.Project
+
+fun String.appendStringByMatch(match: String, append: String, indentation: String, project: Project): String =
+    appendStringByMatch(match, append, append, indentation, project)
+
+/**
+ * Injects a String into a receiver String of content. Before adding it, we check if it will be a duplicate.
+ * We also use another substring in the receiver as the anchor location where we add our new String.
+ *
+ * @receiver The String content into which [append] is being injected.
+ *
+ * @param match The substring we will search for in the receiver content and add [append] to the end of this.
+ * @param appendId If this substring exists in the receiver content already, we won't add [append].
+ * @param append The String we want to add to the receiver content.
+ * @param indentation The indentation to add between [match] and [append].
+ * @param project Temporarily here to show notifications for debugging purposes. Will be removed.
+ * @return String with [append] injected into it, or the same string if no changes are made.
+ */
+fun String.appendStringByMatch(
+    match: String,
+    appendId: String,
+    append: String,
+    indentation: String,
+    project: Project
+): String {
+    val appendIdRegex = appendId.toRegex()
+    val appendIdMatch = appendIdRegex.find(this)?.range
+
+    showNotification(project, "appendId string: $appendId match: $appendIdMatch")
+    return if (appendIdMatch == null) {
```

Comment:
This would be easier to read as early returns, instead of a large nested if-else block. Smaller if-else blocks can make code easier to read than some other ways, but this one is pretty long.